### ISSUE TITLE
Expo Client crashes on iOS 13.4 when toggling Performance Monitor

### ISSUE THREAD

**Issue by darrylyoung** [expo/expo#7469]
## 🐛 Bug Report

First off, I'm not sure if this is the right place to report issues with the client itself but I can't find anywhere else.

Since updating my iPhone (and iOS Simulator) to iOS 13.4, the Expo Client crashes when I try to toggle Performance Monitor. I can reproduce this on my iPhone (iOS 13.4, iPhone XS) and MacBook (macOS 10.15.4, MacBook Pro 16").

### Environment

```
Expo CLI 3.15.5 environment info:
  System:
    OS: macOS 10.15.4
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 12.16.1 - ~/.n/bin/node
    Yarn: 1.22.4 - ~/.yarn/bin/yarn
    npm: 6.14.3 - ~/.n/bin/npm
  IDEs:
    Xcode: 11.4/11E146 - /usr/bin/xcodebuild
  npmPackages:
    expo: ^36.0.0 => 36.0.0
    react: 16.9.0 => 16.9.0
    react-native: https://github.com/expo/react-native/archive/sdk-36.0.0.tar.gz => 0.61.4
  npmGlobalPackages:
    expo-cli: 3.15.5
```

App target: **iOS**

### Steps to Reproduce

- Trigger the Expo menu in the Expo Client (by shaking or two-finger pressing)
- Tap Show Performance Monitor

To be sure it wasn't anything related to code, I reverted to some working code from yesterday and the issue persists.

### Expected Behavior

The Performance Monitor should show in the top corner as it always does

### Actual Behavior

The Expo Client app completely crashes you're taken back to the iOS home screen. I've tested this on both macOS via the iOS Simulator and on iOS on my iPhone.

### Reproducible Demo

I'm not able to share any code from my project but if I get chance after work, I'll try to strip out as much as I can on another branch and create a snack or something. For now, though, it seems unrelated to any code as it was working this morning and since upgrading to the latest macOS and iOS, the Expo Client crashes when toggling the Performance Monitor.

**Comment by IjzerenHein** [comment]
Hi! 👋 I've taken a closer look and want to share an update on what I've found so far.

This appears to be an issue in the react-native code in combination with some changes to JavaScriptCore in iOS 13.4. When enabling the Performance monitor, the `RCTJSCSetOption` function is called to enable GC logging in JavaScriptCore. This throws an exception as of iOS 13.4.

Offending function:

```objc
static BOOL RCTJSCSetOption(const char *option)
{
  static RCTJSCSetOptionType setOption;
  static dispatch_once_t onceToken;

  dispatch_once(&onceToken, ^{
    /**
     * JSC private C++ static method to toggle options at runtime
     *
     * JSC::Options::setOptions - JavaScriptCore/runtime/Options.h
     */
    setOption = dlsym(RTLD_DEFAULT, "_ZN3JSC7Options9setOptionEPKc");

    if (RCT_DEBUG && setOption == NULL) {
      RCTLogWarn(@"The symbol used to enable JSC runtime options is not available in this iOS version");
    }
  });

  if (setOption) {
    return setOption(option); // <-- This throws EXC_BAD_ACCESS exception
  } else {
    return NO;
  }
}
```

I will report this upstream to RN as well and keep you informed of the progress.

For now, I guess you can't use the Performance monitor in iOS 13.4, so you'll need to downgrade if you want to use it.

**Comment by darrylyoung** [comment]
Hi, @IjzerenHein! 👋

Thanks, as always, for your support here. I get that this is out of your control and that's fine. Hopefully we'll get a fix upstream at some point fairly soon. Have a nice day and weekend.

**Comment by IjzerenHein** [comment]
Upstream issue: https://github.com/facebook/react-native/issues/28414

**Comment by IjzerenHein** [comment]
@darrylyoung Yep lets see how this issue develops. I've reported it upstream to RN with a detailed description. It seems that setting options in the JavaScriptCore is no longer supported after the VM has been initialized. So a different strategy is needed if we still want to support VM GC tracking.

**Comment by nextriot** [comment]
Now that it's seemingly been fixed upstream, what's required to get it working in Expo?

**Comment by hadnet** [comment]
Same here. My app crashes when toggling perf monitor, but I'm using a bare workflow of Expo. Any updates on this?

**Comment by IjzerenHein** [comment]
This will be released in a minor SDK update for 37, within the next week or so.

**Comment by darrylyoung** [comment]
> This will be released in a minor SDK update for 37, within the next week or so.

Thanks for the update, @IjzerenHein.

**Comment by IjzerenHein** [comment]
Fixed in Expo client 2.15.5 (iOS)
